packages/protocol/test/governance/network/blockchainparams.ts

```diff
@@ -141,94 +141,95 @@ contract('BlockchainParameters', (accounts: string[]) => {
+        await blockchainParameters.setUptimeLookbackWindow(initialValue)
+        await mineBlocks(EPOCH, web3)
+      })
 
-      await blockchainParameters.setUptimeLookbackWindow(40)
+      it('should set the value for the next epoch', async () => {
+        await blockchainParameters.setUptimeLookbackWindow(newValue)
 
-      // still old value
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+        // still old value
+        assert.equal(
+          (await blockchainParameters.getUptimeLookbackWindow()).toNumber(),
+          initialValue
+        )
 
-      // wait an epoch to find new value
-      await mineBlocks(EPOCH, web3)
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 40)
-    })
+        // wait an epoch to find new value
+        await mineBlocks(EPOCH, web3)
+        assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), newValue)
+      })
 
-    it('if unset should set the variable for the current epoch', async () => {
-      await blockchainParameters.setUptimeLookbackWindow(24)
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
-    })
-    it('if unset, second set should set the variable for the next epoch', async () => {
-      await blockchainParameters.setUptimeLookbackWindow(24)
-      await blockchainParameters.setUptimeLookbackWindow(50)
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
-      await mineBlocks(EPOCH, web3)
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 50)
-    })
+      it('multiple calls within epoch only applies last', async () => {
+        // make 2 calls
+        await blockchainParameters.setUptimeLookbackWindow(newValue)
+        await blockchainParameters.setUptimeLookbackWindow(50)
 
-    it('when doing 2 sets, should override the value for the next epoch', async () => {
-      // enter steady state
-      await blockchainParameters.setUptimeLookbackWindow(24)
-      await mineBlocks(EPOCH, web3)
+        // current epoch still initial value
+        assert.equal(
+          (await blockchainParameters.getUptimeLookbackWindow()).toNumber(),
+          initialValue
+        )
 
-      await blockchainParameters.setUptimeLookbackWindow(lookbackWindow)
-      await blockchainParameters.setUptimeLookbackWindow(50)
+        // after the epoch, find last value set
+        await mineBlocks(EPOCH, web3)
+        assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 50)
+      })
+    })
 
-      // current epoch still initial value
-      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 24)
+    describe('when not initialized', () => {
+      it('should set the variable for the current epoch', async () => {
+        await blockchainParameters.setUptimeLookbackWindow(initialValue)
+        assert.equal(
```

Comment:
You'd probably technically want to make this assertion before setting in case initialValue === 12

Reply:
There's already a tests for that:
```ts
    it('should return 12 if not initalized', async () => {
      assert.equal((await blockchainParameters.getUptimeLookbackWindow()).toNumber(), 12)
    })
```

Reply:
What I'm saying is that if `initialValue` is 12, then a faulty setUptimeLookbackWindow (i.e. set in the next epoch) would still make this test pass,right?

Reply:
Do you mean before setting to have the same assert using `.notEqual` to ensure the before and after values can't coincidentally be equal?